controllers/datafoundation.go

```diff
@@ -271,24 +272,34 @@ func (r *dataFoundationReconciler) reconcileStorageCluster() error {
 			return err
 		}
 
-		// Convert the desired size to the device set count based on the underlaying OSD size
-		desiredDeviceSetCount := int(math.Ceil(float64(r.spec.UsableCapacityInTiB) / templates.OSDSizeInTiB))
+		UsableCapacityInTiB := resource.MustParse(fmt.Sprintf("%dTi", r.spec.UsableCapacityInTiB))
+		OSDSizeUpperBoundInTiB := resource.MustParse(fmt.Sprintf("%dTi", templates.OSDSizeUpperBoundInTiB))
 
-		// Get the storage device set count of the current storage cluster
-		currDeviceSetCount := 0
-		if desiredStorageDeviceSet := findStorageDeviceSet(r.storageCluster.Spec.StorageDeviceSets, defaultDeviceSetName); desiredStorageDeviceSet != nil {
-			currDeviceSetCount = desiredStorageDeviceSet.Count
+		// Get the desired capacity in TiB planned to be consumed by user in range 1-4TiB
+		// Set desired to be the minimum
+		desiredOSDSizeInTiB := UsableCapacityInTiB
+		if OSDSizeUpperBoundInTiB.Cmp(UsableCapacityInTiB) < 0 {
+			desiredOSDSizeInTiB = OSDSizeUpperBoundInTiB
 		}
 
+		// Convert the desired size to the device set count based on the underlying OSD size
+		desiredDeviceSetCount := int(math.Ceil(UsableCapacityInTiB.AsApproximateFloat64() / desiredOSDSizeInTiB.AsApproximateFloat64()))
```

Comment:
No need to transform back to bytes and lose precision, you can use `ScaledValue` as a better alternative.
```suggestion
		desiredDeviceSetCount := int(math.Ceil(
			float(UsableCapacityInTiB.ScaledValue("Tera")) / float(desiredOSDSizeInTiB.ScaledValue("Tera"))
		))
```

Reply:
@raaizik This is not a must, if you feel the code is good as is we can merge it as is

Reply:
@nb-ohad At some point in the future we would probably need to support non-tera values, no?

Reply:
Storage provisioning for an entire cluster (not a volume provided by the cluster) is provisioned at the minimum GiBs levels, but in most cases at the TiB level. I would even make a case that our code should prevent allowing clusters to be provisioned at a unit below the GiB unit. 

If you feel that you prefer to translate everything to GiBs, I am ok with that. The main thing is that I would like to eliminate the imprecisions that come with translating to very big floats (in this case, translating to bytes)